### PR TITLE
Parse escaped chars

### PR DESCRIPTION
**Pull request by schorschii**
Hi! First of all, thanks for this library!

I noticed that it does not parse escaped chars like `\n` or `\u00fc` (ü), see [here](https://stackoverflow.com/questions/19176024/how-to-escape-special-characters-in-building-a-json-string#answer-27516892).

This change adds support for those escaped values.

### CODE REVIEW

File: Source/JSON.swift [hunk 1]
```diff
@@ -596,7 +596,29 @@ open class JSON {
...
         }
         
-        let parsedString = String(jsonString[startingIndex ..< index])
+        var parsedString = String(jsonString[startingIndex ..< index])
+            .replacingOccurrences(of: "\\\\", with: "\\", range: nil)
+            .replacingOccurrences(of: "\\n", with: "\n", range: nil)
+            .replacingOccurrences(of: "\\t", with: "\t", range: nil)
+            .replacingOccurrences(of: "\\r", with: "\r", range: nil)
+            .replacingOccurrences(of: "\\\"", with: "\"", range: nil)
+        if let regex = try? NSRegularExpression(pattern: "\\\\u[0-9A-Fa-f]{4}") {
```
Comment (mikezs) [inline Source/JSON.swift]: What's this block doing, it's not very descriptive and doesn't look like it relates to escaped characters, it looks like some sort of hex number parsing.

Edit: I see this is from the link you posted in the original PR, but the variable names don't make it obvious at all

File: Source/JSON.swift [hunk 2]
```diff
@@ -596,7 +596,29 @@ open class JSON {
             }
         }
         
-        let parsedString = String(jsonString[startingIndex ..< index])
+        var parsedString = String(jsonString[startingIndex ..< index])
```
Comment (mikezs) [inline Source/JSON.swift:607]: This could stay as a constant and be given a different name and built upon, e.g. 
`let initialParsedString = String()`
`let parsedString = initialParsedString.someOperation()`

**Comment by mikezs** [comment]
I've added some suggestions, and please add some tests for each of the scenarios and edge cases please

**Comment by schorschii** [comment]
I added a test for `\n` and `\u00fc` and also fixed the `testEscapedQuoteString()` test:

```

let input = """
["\\\""]
"""
```
results in `["\""]`, which is just a `"` when decoding the JSON string. The previous version expected `\"` which is wrong.


> I've added some suggestions

What do you mean? I can't see any comments or changes on the code.

**Comment by mikezs** [comment]
> What do you mean? I can't see any comments or changes on the code.

They're comments on the code made on this pull request, see here: https://github.com/mikezs/Tisander/pull/2

**Comment by schorschii** [comment]
I'm sorry but I can't see any comments.

**Review by mikezs — CHANGES_REQUESTED** [review]
Sorry, apparently I left this in draft

File: Source/JSON.swift (hunk 2 above)
Comment (schorschii) [reply Source/JSON.swift:607]: I'm not sure if this is useful. I mean how would you implement this for the `hexEncodedUnicodeCharRegex` `parsedString`? Feel free to push your solution on this branch.
